src/v/cluster/cluster_utils.h

```diff
@@ -25,6 +25,23 @@
 
 #include <utility>
 
+namespace detail {
+
+template<typename T, typename Fn>
+std::vector<cluster::topic_result>
+create_topic_results(const std::vector<T>& topics, Fn&& fn) {
```

Comment:
it doesn't look like you need a forwarding reference for `fn`. am I missing something?